virttest/storage.py

```diff
@@ -68,20 +68,31 @@ def file_exists(params, filename_path):
 
     :return: True if image file exists else False
     """
-    gluster_image = params.get("gluster_brick")
-    if gluster_image:
-        return gluster.file_exists(params, filename_path)
-
-    if params.get("enable_ceph") == "yes":
+    file_existed = False
+    if params.get("gluster_brick"):
+        file_existed = gluster.file_exists(params, filename_path)
+    elif params.get("enable_ceph") == "yes":
         image_name = params.get("image_name")
         image_format = params.get("image_format", "qcow2")
         ceph_monitor = params.get("ceph_monitor")
         rbd_pool_name = params["rbd_pool_name"]
         rbd_image_name = "%s.%s" % (image_name.split("/")[-1], image_format)
         ceph_conf = params.get("ceph_conf")
-        return ceph.rbd_image_exist(ceph_monitor, rbd_pool_name,
-                                    rbd_image_name, ceph_conf)
-    return os.path.exists(filename_path)
+        ceph_key = params.get("ceph_key")
```

Comment:
Not sure if we would support variants of forms, such as passing a keyring file or giving the description of keys, and if so the data structure should be more complex. And even if we only support the latter, it still needs a change since sometimes users may want to specify the entity name of caps. Overall, I'd suggest using json format to represent this value.

Reply:
As above.